src/ScottPlot5/ScottPlot/Control/InputMap.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace ScottPlot.Control;
+
+/// <summary>
+/// This class maps common plot actions with specific mouse and keyboard inputs
+/// </summary>
+public class InputMap
+{
+    public MouseButton? ClickDragPan = null;
+    public MouseButton? ClickDragZoom = null;
+    public MouseButton? ClickDragZoomRectangle = null;
+    public MouseButton? DoubleClick = null;
+    public MouseButton? ZoomIn = null;
+    public MouseButton? ZoomOut = null;
+
+    public Key? LockHorizontalAxis = null;
+    public Key? LockVerticalAxis = null;
+    public Key? PanZoomRectangle = null;
+
+    public bool IsLockedX(IEnumerable<Key> keys)
```

Comment:
This method (and `IsLockedY` and `IsZoomingRectangle`) sound like they're stateful. i.e. They're storing whether or not the axis should be locked or the rectangle should be zooming.

Perhaps it should be `ShouldLockX`, `ShouldLockY` and `ShouldZoomRectangle`?

Reply:
Good point! I'll do this, and improve understanding using XML docs too 👍